### ISSUE TITLE
Ready for WCAG 2.2 ♿ 🏖️ - Core

### ISSUE THREAD

**Issue by sstrubberg** [carbon-design-system/carbon#15592]
### User story

> As a `IBM Maker`:

> I need to: build products that meet WCAG 2.2 criteria

> so that I can:  close deals where this is a requirement *and* we build as inclusive of experience as possible for all users.

### Additional information

- _https://www.w3.org/TR/WCAG22/_
- _https://www.w3.org/TR/WCAG22/#new-features-in-wcag-2-2_
- _https://design-system.service.gov.uk/accessibility/wcag-2.2/_
- _[Conversation in CAG](https://ibm-studios.slack.com/archives/G01GCBCGTPV/p1705939267734899)_

### Acceptance criteria

- [ ] _All necessary components meet WCAG 2.2_
- [ ] _All website documentation meet WCAG 2.2_
- [ ] _EAC CI/CD instrumented accordingly

### Mike Gower's review reference note
- https://ibm.ent.box.com/s/e2hhcmw1u6teofxgqx2l1nlw89mbxii0

### Tasks
- [x] Determine which components are effected by said changes above (audit)
- [ ] Update components effected
- [ ] Update website docs effected
- [ ] Update kits effected
- [ ] Update EAC accordingly
- [x] https://github.com/carbon-design-system/carbon/issues/12594
- [ ] Validate Risk Analysis

**Comment by sstrubberg** [comment]
![image (5)](https://github.com/carbon-design-system/carbon/assets/15822070/3e564d9b-96e3-46fa-92b6-341348ca1b10)

**Comment by dav-idc** [comment]
Hey Carbon Design System team! I worked on the recent GOV.UK Design System WCAG 2.2 updates, along with the rest of the team. Up until November, I was the senior accessibility specialist and project manager for the WCAG 2.2 updates.

Let me know if you have any questions about the updates we made, or about how WCAG 2.2 interacts with design system components and guidance documentation!

Related links:
- [GOV.UK Design System WCAG 2.2 updates page](https://design-system.service.gov.uk/accessibility/wcag-2.2/)
- [Blog post announcing the WCAG 2.2 updates](https://accessibility.blog.gov.uk/2024/01/11/get-to-wcag-2-2-faster-with-the-gov-uk-design-system/)
- [GitHub 'epic' issue for the WCAG 2.2 work](https://github.com/alphagov/govuk-frontend/issues/3040)

**Comment by sstrubberg** [comment]
@mbgower's reference notes https://ibm.ent.box.com/s/e2hhcmw1u6teofxgqx2l1nlw89mbxii0

**Comment by sstrubberg** [comment]
Beef up the design guidance.
